tools/websocket_to_posix_proxy/src/main.cpp

```diff
@@ -0,0 +1,382 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <memory.h>
+#include <sys/types.h>
+#include "posix_sockets.h"
+#include "threads.h"
+#include <assert.h>
+#include <vector>
+
```

Comment:
It seems that `signal.h` is missing here for at least `__linux__`. I assume `__APPLE__` uses the same header, but I don't know.